### PR TITLE
Fixing numerous issues discovered whilst developing new event source 

### PR DESCRIPTION
**Pull request by jthomas**
Whilst developing a new event source (https://github.com/jthomas/openwhisk-amqp-trigger-feed) for the pluggable provider, I discovered a few bugs and other issues to resolve upstream in the provider code. 

- Parameter validation errors were failing to be re-thrown causing invalid triggers to be added.
- Replace `var` with `const/let` for defensive coding .
- Clearing out more references to "Cloudant" from the original fork'd project code.
- Make it easier to run in development using the `EVENT_PROVIDER_LIB` variable.

### CODE REVIEW

**Review by rabbah — APPROVED** [review]

File: actions/event-actions/changesWebAction.js [hunk 1]
```diff
@@ -36,7 +36,7 @@ function main(params) {
           common.verifyTriggerAuth(triggerData, false)
             .then(() => EventProvider.validate(params))
             .catch(err => {
-              return reject(common.sendError(400, `Feed parameter validation failed`, err.message));
+              throw common.sendError(400, `Feed parameter validation failed`, err.message);
```
Comment (markusthoemmes) [inline actions/event-actions/changesWebAction.js:39]: Shouldn't this actually be what you had above? `Promise.reject(...)`?

File: provider/lib/triggers_manager.js [hunk 2]
```diff
@@ -47,81 +47,6 @@ module.exports = function(logger, triggerDB, EventProvider, redisClient) {
...
-                });
-            });
-
-        } catch (err) {
-            logger.info(method, 'caught an exception for trigger', triggerData.id, err);
-            return Promise.reject(err);
-        }
-
-    };
-*/
```
Comment (markusthoemmes) [inline provider/lib/triggers_manager.js:123]: Remove this code vs. commenting it out?

**Review by markusthoemmes — COMMENTED** [review]
Two nits.

File: actions/event-actions/changesWebAction.js (hunk 1 above)
Comment (jthomas) [reply actions/event-actions/changesWebAction.js:39]: The existing code returns the result of the `reject` function which is a non-error value and causes the following `then` handlers to run (which is not what we want). By throwing the error, all the `then` handlers are skipped and the `catch` at the end can reject the returned promise with the error from the validation issue. Did you spot something else?

File: provider/lib/triggers_manager.js (hunk 2 above)
Comment (jthomas) [reply provider/lib/triggers_manager.js:123]: I'm removing code that was commented out as it was unused from the re-working on the codebase. I didn't mean to leave this in originally. Is that what you mean?

File: actions/event-actions/changesWebAction.js (hunk 1 above)
Comment (markusthoemmes) [reply actions/event-actions/changesWebAction.js:39]: If you return `Promise.reject` that should not happen. Javascript doesn't care (or even know) what error values are, as long as the promise is rejected underneath.

**Pull request merged**